### DISCUSSION TITLE
Pinecil v2: Rename Bluetooth option or keep BLE?

### DISCUSSION THREAD

**Discussion by StevenSeifried** [Ralim/IronOS#1621] [Polls]
Hey there,

I'm considering whether it makes sense to rename the Bluetooth option from BLE to Bluetooth or Bluetooth LE.
I think that the most users know what BLE mean, but Bluetooth or Bluetooth LE are more wildly used.

What do you think? Depending on the results of this survey, I will open a pull request (or not).

#### Current
![current](https://user-images.githubusercontent.com/39765956/224758417-50d9fe56-2ca1-4570-af89-573ca2cf4e50.jpg)

#### Bluetooth
![Bluetooth](https://user-images.githubusercontent.com/39765956/224758657-7ca03d06-d4a9-4bf0-b10c-aa40fc6d4ad7.jpg)

#### Bluetooth LE
![Bluetoorh_LE](https://user-images.githubusercontent.com/39765956/224758692-0170d31e-09c3-4e52-933c-deef5a2430c7.jpg)

**Comment by River-Mochi** [comment]
what about also adding the pixel art for Bluetooth symbol? several examples already exist,  it would be best in just black & white.
https://www.pinterest.com/pin/1091771134640308034/
![image](https://user-images.githubusercontent.com/97197236/225159300-fc65109f-37c8-4c4e-a629-e516ccffea9d.png)

needs to be a lot smaller than that. I'm not sure of the parameters for  square icon on IronOS.

@discip   do you know offhand  if  Bluetooth pixel icon could be done there?

**Reply by discip** [reply]
I could try.

**Reply by discip** [reply]
![BLE](https://user-images.githubusercontent.com/53649486/225161037-3a8bd90f-3baa-4dff-b549-ac54e1e34146.png)

**Reply by River-Mochi** [reply]
> ![BLE](https://user-images.githubusercontent.com/53649486/225161037-3a8bd90f-3baa-4dff-b549-ac54e1e34146.png)

Dicip this will also be good for non-english readers since that symbol is universal. needs no translation.

**Reply by discip** [reply]
I agree, but unfortunately this is a very memory-hungry solution if we were to implement it throughout the firmware.

Please see the comment below for more background information.

**Reply by River-Mochi** [reply]
> I agree, but unfortunately this is a very memory-hungry solution if we were to implement it throughout the firmware.
> 
> Please see the comment below for more background information.

I would not impement it  on all the irons.  there is no need since  there is no bluetooth on anything else. 
@discip   the T80 and T80p  do not have a Bluetooth category in  advanced menu  correct?  so how would a BLuetooth icon  affect  them?  They also do not have a lot of other features that Pinecil V2 has  like  all the extra code for  detecting short and long  ts100 tips since they don't support those tips. 
I'm confused how   a T80 not having memory is related to Bluetooth.  Could you  explain?

**Comment by River-Mochi** [comment]
@StevenSeifried  I vote for the Bluetooth Icon that Dicip just made. with words BLE.

but  I don't mind if it's also the icon and bluetooth word or BLE word .  Must have the Art Icon though as this is Universal Symbol!! 
would be fantastic if the Icon  art comes out with the 2.21  release as that will be first time that  we have BLE  onboard a stable release after ~6 months of testing.

**Comment by builder555** [comment]
i like pixel art

**Comment by StevenSeifried** [comment]
I had not thought of a Bluetooth symbol. It is a great idea. Thanks @River-Mochi 

Thanks @discip for the icon. It's great 

What do you think about \
`Symbol | BLE Enabled | Checkbox`? 
(`Symbol | Bluetooth Enabled | Checkbox`) should also fit on the screen

**Reply by River-Mochi** [reply]
Actually, Dancho brought it up in chat, I just made the effort to find one on search engine to use.
not a fan of the "LE".  
either of these two would work now that we have International Symbol. 

Icon | bluetooth  [ ]   
Icon | BLE  [ ]

**Reply by discip** [reply]
If you ask me:
The text is not necessary if you have an icon!

This also worked with "screen brightness" and "invert screen" some time ago!

Users didn't like having a mix of **_icons_** and **_text_**.

I can't remember where, but at some point there was a suggestion to have a GUI based on icons rather than text, which I would definitely prefer over the current one, but unfortunately the **`TS80P`** has the smallest memory.
So unfortunately a _no go_.

I could try to do that for **`Pinecil V1/2`** but to be able to test all the features including `BLE` I would need a `PinecilV2` which unfortunately I don't have. 😊

**Reply by River-Mochi** [reply]
> I could try to do that for **`Pinecil V1/2`** but to be able to test all the features including `BLE` I would need a `PinecilV2` which unfortunately I don't have. 😊

_Only_ the Pinecil V2  has bluetooth and it has a lot of memory.   this feature would not go onto a T80P model ever.
I don't think it's an issue or related since the V2 is unique and the only device with  Bluetooth.

 All the BLE  code is **only** for Pinecil V2 and not even the V1 since it doesn't have it. 

if  you make both  it in beta  format  **I can test it  both ways with  at least 8 other people**  right away that I know  in volunteer  Pine64 chat.  we could test the Icon + BLE  [ x]  box  and the  Icon + bluetooth [ x]
and see what the user experience is. 

I don't think we have to worry about any BLE code as a comparison for any other Miniware or even the Pinecil V1 since it's not relevant the old models dont have bluetooth. 

in the Pine64 chat we also have people from all over the world who help to test  Pinecil things so it's also a mix of people from different languages who test. 

**Side note** - we have several people who code for Pinecil without having a Pinecil.
for example Spagett made an entire Linux  Gui updater for Pinecil V1  and he never had a V1.  what he did  was had a bunch of us test the code and give him feedback.  we remotely tested features and gave him daily feedback.  he did it for fun and bc he is part of the pine64 community.  and now we have a fully functional Linux Gui based  installer for Pinecil V1 since people still have those (they now don't need to use dfu command line with it)

Another person worked on  getting BLE  to work on MacOS and iphone  for Pinecil V2 and he owns  no mac and no  iphone but he knows about BLE a lot.  basically he set up some tester webpages and   people in Pine64 chat  volunteered to do testing with him a  1-2 hours a day.  we used our iphones  and  macbooks, went to website   he set up with Pinecil V2  on  and then paired with  website and gave him  the output  from his  debugging page.
note  he  aso did all of this with no V2 either since  he only got it 1 day ago.

Basically it's totally fine if you don't have a V2,  it's not necessary at all.    it's just a matter of adding the icon symbol  to the BLE text that is already there, and then  we can test it.  or a version with Icon and Bluetooth word.

or @discip   just put the icon there and @StevenSeifried   could add the words to two different betas and give it to us to test.

**Reply by StevenSeifried** [reply]
When it comes to testing, I also make myself available. For all Pinecil v2 testing things (if I'm able to test it)

I'm not often on Discord, but over GitHub is no problem.

**Reply by River-Mochi** [reply]
I half agree with dicip and half with stephen . The pixel art will take space on screen 
More than one character wide.  might end up being art+ ble and box [x]

I'm willing to test any of them but getting icon on blisp is something I don't know how do yet . Need ralim or discip to get ble icon in there

**Reply by gamelaster** [reply]
@discip can you please contact me on Discord (GAMELASTER#8007) or EMAIL (gamiee@pine64.org)? Thanks!

**Reply by discip** [reply]
Will do later, when I'm home again.

**Reply by StevenSeifried** [reply]
Hi,
is there any news regarding the BLE symbol implementation?

**Reply by River-Mochi** [reply]
on newest beta I just installed tonight with "firmware version sent on Ble" it shows this but no Bluetooth icon yet.
Having icon would be nice since it's universal ![image](https://user-images.githubusercontent.com/97197236/227444627-e4fbada3-e12e-400a-aaff-54f50be317a7.jpeg)

**Comment by River-Mochi** [comment]
need to wait for dicip to add Icon to Beta, and then  could see what could fit there with Icon and the checkbox  as must have.

**Comment by River-Mochi** [comment]
I opened an Issue ticket to get this moving 
https://github.com/Ralim/IronOS/issues/1640

Please thumbs up or comment on the issue ticket
